drivers/modem/ublox-sara-r4.c

```diff
@@ -1258,6 +1348,41 @@ static int create_socket(struct modem_socket *sock, const struct sockaddr *addr)
 		return -1;
 	}
 
+	if (sock->ip_proto == IPPROTO_TLS_1_2) {
+		char buf[sizeof("AT+USECPRF=#,#,#######\r")];
+
+		snprintk(buf, sizeof(buf), "AT+USOSEC=%d,1,%d", sock->id, sock->id);
+		modem_cmd_send(&mctx.iface, &mctx.cmd_handler, NULL, 0U, buf, &mdata.sem_response,
+			       MDM_CMD_TIMEOUT);
+		/* Reset the security profile */
+		snprintk(buf, sizeof(buf), "AT+USECPRF=%d", sock->id);
+		modem_cmd_send(&mctx.iface, &mctx.cmd_handler, NULL, 0U, buf, &mdata.sem_response,
+			       MDM_CMD_TIMEOUT);
+		/* Disable server cert validation TODO:Validation level should be at least 1*/
+		snprintk(buf, sizeof(buf), "AT+USECPRF=%d,0,0", sock->id);
+		modem_cmd_send(&mctx.iface, &mctx.cmd_handler, NULL, 0U, buf, &mdata.sem_response,
+			       MDM_CMD_TIMEOUT);
+		/* Use TLSv1.2 only */
+		snprintk(buf, sizeof(buf), "AT+USECPRF=%d,1,3", sock->id);
+		modem_cmd_send(&mctx.iface, &mctx.cmd_handler, NULL, 0U, buf, &mdata.sem_response,
+			       MDM_CMD_TIMEOUT);
+		/* Set root CA filename */
+		snprintk(buf, sizeof(buf), "AT+USECPRF=%d,3,\"ca\"", sock->id);
```

Comment:
This looks like another limitation - only a single root CA is supported, while the TLS sockets allow configuring PSK security as well. Should be listed somewhere as well.